pkg/console/controllers/route/controller.go

```diff
@@ -159,6 +186,48 @@ func (c *RouteSyncController) SyncRoute(operatorConfig *operatorsv1.Console) (co
 	return rt, rtIsNew, "", rtErr
 }
 
+// Custom route sync needs to:
+// 1. validate if the reference for secret with serving-cert is defined in operator config(in case a non-openshift cluster domain is used)
+// 2. sync the secret with the serving-cert, from the openshift-config,into the openshift-console namespace
+// 3. create the custom console route
+// 4. check custom route health
+// 5. validate the route
+func (c *RouteSyncController) SyncCustomRoute(operatorConfig *operatorsv1.Console) (consoleRoute *routev1.Route, isNew bool, reason string, err error) {
+	if configErr := routesub.ValidateCustomRoute(operatorConfig, c.routeClient); configErr != nil {
+		return nil, true, "InvalidCustomRouteConfig", configErr
+	}
+
+	if scscErr := c.SyncCustomServigCertSecret(operatorConfig); scscErr != nil {
+		return nil, false, "FailedSyncServingCertSecret", scscErr
+	}
+
+	cr, crIsNew, crErr := routesub.GetOrCreate(c.routeClient, routesub.CustomRoute(operatorConfig))
+	if crErr != nil {
+		return nil, false, "FailedCreateCustomRoute", crErr
+	}
+	c.CheckRouteHealth(operatorConfig, cr)
+
+	if _, changed := routesub.Validate(cr); changed {
+		return nil, true, "InvalidCustomRoute", customerrors.NewSyncError("custom route is invalid, correcting route state")
+	}
+
+	return cr, crIsNew, "", crErr
+}
+
```

Comment:
This should be good to update the resource sync source.

Reply:
you mean that here we should call the resource sync for syncing the custom Secret with serving-cert?

Reply:
oh just meant looks good to me.
positive comment, no issues :)